src/ol/source/Vector.js

```diff
@@ -544,7 +544,10 @@ class VectorSource extends Source {
       }
     } else {
       if (this.featuresRtree_) {
-        this.featuresRtree_.forEach(this.removeFeatureInternal.bind(this));
+        // use Array forEach to ignore return
+        this.featuresRtree_
+          .getAll()
```

Comment:
This makes things slower, and I do not see a need for it. What was the rationale behind this change?

Reply:
The Rtree f`orEach` method calls `getAll()`
https://github.com/openlayers/openlayers/blob/main/src/ol/structs/RBush.js#L154
but exits if the callback returns a truthy value (i.e. the feature)
https://github.com/openlayers/openlayers/blob/main/src/ol/structs/RBush.js#L178
That broke the `.clear()` test https://github.com/openlayers/openlayers/blob/main/test/browser/spec/ol/source/vector.test.js#L278 as although all the features were removed only one `removefeature` event was fired.  It is only going to be slower if Array `forEach` is slower than a `for` loop.

Reply:
Oh no, that `structs/RBush` wrapper could use some love.

But thanks for the explanation! This means adding a return value to `removeFeatures()` could be a breaking change. Is the return value really useful?

Reply:
Yes it would break `map.forEachFeatureAtPixel(pixel, source.removeFeature.bind(source));`  I have never used the return values from the other remove methods so maybe that could wait until a full release,